.github/workflows/ai_pr_title_review.yml

```diff
@@ -24,21 +23,10 @@ jobs:
           egress-policy: audit
 
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
if we are using pull_request_target lets change checkout to pull only the base branch (develop) with
ref: ${{ github.event.pull_request.base.ref }}
There isn't really a need to pull external branch since the diff is done via URL